### ISSUE TITLE
Borrow checker: incorrect error of used of moved value.

### ISSUE THREAD

**Issue by kristof-mattei** [intellij-rust/intellij-rust#8659]
## Environment

* **IntelliJ Rust plugin version:** 0.4.166.4486-213
* **Rust toolchain version:** 1.59.0 (9d1b2106e 2022-02-23) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2021.3.3 (CL-213.6777.58)
* **Operating system:** Windows 11 10.0
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description
Incorrect error that we're using a moved value. 

## Steps to reproduce
<img width="597" alt="image" src="https://user-images.githubusercontent.com/864376/158038044-29b4dae0-f6ab-4255-90b4-9c8d1dc8f294.png">

```rust
pub struct Quz {
    pub next: String,
}

fn bar(str: String) -> String {
    str
}

fn main() {
    let mut quz = Box::new(Quz {
        next: "foo".to_string(),
    });

    let new_next = bar(quz.next);

    quz.next = new_next;

    println!("{}", &quz.next);
}
```
